tests/test_add_fund_and_claim.cairo

```diff
@@ -0,0 +1,418 @@
+
+    let mut guilds: Array<felt252> = ArrayTrait::new();
+    guilds.append('dev');
+    guilds.append('design');
+    guilds.append('problem_solving');
+    guilds.append('marcom');
+    guilds.append('research');
+
+    let mut amounts1: Array<u256> = ArrayTrait::new();
+    amounts1.append(10 * TOKEN_MULTIPLIER);
+    amounts1.append(20 * TOKEN_MULTIPLIER);
+    amounts1.append(30 * TOKEN_MULTIPLIER);
+    amounts1.append(40 * TOKEN_MULTIPLIER);
+    amounts1.append(50 * TOKEN_MULTIPLIER);
+
+    let mut guilds2: Array<felt252> = ArrayTrait::new();
+    guilds2.append('research');
+
+    let mut amounts2: Array<u256> = ArrayTrait::new();
+    amounts2.append(40 * TOKEN_MULTIPLIER);
+
+    start_prank(usdc, deployer_addr());
+    usdc_dispatcher.approve(salary_address, 200 * TOKEN_MULTIPLIER);
+    stop_prank(usdc);
+
+    let month_id = 092023;
+    start_prank(salary_address, deployer_addr());
+    safe_salary_dispatcher.add_fund_to_salary_pools(month_id, amounts1, guilds);
+    match safe_salary_dispatcher.add_fund_to_salary_pools(month_id, amounts2, guilds2) {
+        Result::Ok(_) => panic_with_felt252('shouldve panicked'),
+        Result::Err(panic_data) => {
+            assert(*panic_data.at(0) == 'ALREADY_SET', *panic_data.at(0));
+        }
+    };
+    stop_prank(salary_address);
+
+}
+
+#[test]
+fn test_add_fund_and_claim() { 
+    let initial_supply: u256 = 1000 * TOKEN_MULTIPLIER;
+    let usdc = deploy_mockUSDC(initial_supply);
+    let (master_address, salary_address) = deploy_contracts(usdc);
+    update_contributions(master_address);
+
+    let master_dispatcher = IMasterDispatcher { contract_address: master_address };
+    let salary_dispatcher = ISalaryDispatcher { contract_address: salary_address };
+    let usdc_dispatcher = IERC20Dispatcher { contract_address: usdc };
+
+    let safe_salary_dispatcher = ISalarySafeDispatcher { contract_address: salary_address };
+
+    let mut guilds: Array<felt252> = ArrayTrait::new();
+    guilds.append('dev');
+    guilds.append('design');
+    guilds.append('problem_solving');
+    guilds.append('marcom');
+    guilds.append('research');
+
+    let mut amounts1: Array<u256> = ArrayTrait::new();
+    amounts1.append(10 * TOKEN_MULTIPLIER);
+    amounts1.append(20 * TOKEN_MULTIPLIER);
+    amounts1.append(30 * TOKEN_MULTIPLIER);
+    amounts1.append(40 * TOKEN_MULTIPLIER);
+    amounts1.append(50 * TOKEN_MULTIPLIER);
+
+    let mut amounts2: Array<u256> = ArrayTrait::new();
+    amounts2.append(50 * TOKEN_MULTIPLIER);
+    amounts2.append(40 * TOKEN_MULTIPLIER);
+    amounts2.append(30 * TOKEN_MULTIPLIER);
+    amounts2.append(20 * TOKEN_MULTIPLIER);
+    amounts2.append(10 * TOKEN_MULTIPLIER);
+   
+
+    start_prank(usdc, deployer_addr());
+    usdc_dispatcher.approve(salary_address, 300 * TOKEN_MULTIPLIER);
+    stop_prank(usdc);
+
+    start_prank(salary_address, deployer_addr());
+    salary_dispatcher.add_fund_to_salary_pools(092023, amounts1, guilds.clone());
```

Comment:
What's the reason for using `guilds.clone()`? In the previous tests, you are not using `clone()` method. I'm just curious :)

Reply:
when you parse a variable as an input in function, it is moved. So for using the same variable multiple places, we need to clone it.